ideascube/widgets.py

```diff
@@ -1,6 +1,101 @@
+selector: "#{widget_id}",
+inline: true,
+theme: "inlite",
+hidden_input: false,
+menubar: false,
+toolbar: false,
+selection_toolbar: {settings[selection_toolbar]},
+insert_toolbar: {settings[insert_toolbar]},
+plugins: {settings[plugins]}{settings[extra_settings]}
+}});</script>'''
+
+
+class RichTextEntry(widgets.Widget):
+    def __init__(self, *,
+            with_media = False,
+            selection_toolbar=None,
+            attrs=None):
+        super().__init__(attrs=attrs)
+
+        self.authorized_tags = ['p', 'a', 'ul', 'ol', 'li', 'br',
+                                'h2', 'h3', 'h4', 'h5']
+
+        self.authorized_attributes = {
+            'a': ['href', 'title']
+        }
+
+        if not selection_toolbar is False:
+            selection_toolbar = 'false'
+        else:
+            if selection_toolbar in (True, None):
+                selection_toolbar = '"bold italic | quicklink h2 h3 blockquote"'
+            else:
+                selection_toolbar = '"{}"'.format(' '.join(selection_toolbar))
+
+        insert_toolbar = ['bullist', 'numlist']
+        extra_settings = ['']
+        plugins = ['autolink', 'textpattern']
+        if with_media:
+            extra_settings += ['media_url_resolver: ID.image_url_resolver']
+            insert_toolbar += ['media']
+            plugins += ['media']
+            self.authorized_tags += ['img', 'iframe']
+            self.authorized_attributes['img'] = ['src', 'width', 'height',
+                                                 'alt']
+            self.authorized_attributes['iframe'] = ['src', 'width', 'height',
+                                                    'allowfullscreen']
+
+        self.settings = {
+            'selection_toolbar': selection_toolbar,
+            'insert_toolbar': '"{}"'.format(' '.join(insert_toolbar)),
+            'plugins': '"{}"'.format(' '.join(plugins)) ,
+            'extra_settings': ',\n'.join(extra_settings)
+        }
+
+    def render(self, name, value, attrs=None):
+        if value is None:
+            value = ''
+
+        # The widget displayed to the user
+        div_attrs = {'id': name}
+        div_html = '<div{}>{}</div>'.format(
+            flatatt(div_attrs),
+            force_text(value)
+        )
+
+        # The input used by django
+        input_attrs = self.build_attrs(attrs, name=name, value=escape(value),
+                                       type='hidden')
+        input_html = '<input{} />'.format(
+            flatatt(input_attrs)
+        )
+
+        # Make tinymce handle the widget
+        script_html = _script_template.format(
+            widget_id = name,
+            settings = self.settings)
+
+        html = '\n'.join([div_html, input_html, script_html])
+        return mark_safe(html)
+
+    def value_from_datadict(self, data, files, name):
+        raw_html_input = super().value_from_datadict(data, files, name)
+        filter_html = bleach.clean(
```

Comment:
Minor nitpick: this would be better called `filtered_html`, since `filter` is a verb and would better represent a function than a result, the result being filterED.

I won't block the review on this, but if you have a minute to change this then I'll be happier. 😄